modules/cold-extraction/ColdDataRetriever.py

```diff
@@ -161,33 +132,35 @@ def initialize():
     subprocess.call("{0}/storescp --accept-unknown --directory {1} --filepath {2} -b {3} > storescp.out &".format(DCM4CHE_BIN, storage_folder, file_path, QUERY_AET), shell=True)
 
 
-
-with open(csv_file, newline='') as f:
-    reader = csv.reader(f)
-    next(f)
-    for row in reader:
-        row = [x.strip() for x in row]
-        if (extraction_type == 'empi_date'):
-            if not ((row[patient_index] == "") or (row[date_index] == "")):
-                patients.append(row[patient_index])
-                temp_date = row[date_index]
-                dt_stamp = datetime.datetime.strptime(temp_date, date_format)
-                date_str = dt_stamp.strftime('%Y%m%d')
-                dates.append(date_str)
-                length = len(patients)
-        elif (extraction_type == 'empi'):
-            if not ((row[patient_index] == "")):
-                patients.append(row[patient_index])
-                length = len(patients)
-        elif (extraction_type == 'accession'):
-            if not ((row[accession_index] == "")):
-                accessions.append(row[accession_index])
-                length = len(accessions)
-        elif (extraction_type == 'empi_accession'):
-            if not ((row[patient_index] == "") or (row[accession_index] == "")):
-                patients.append(row[patient_index])
-                accessions.append(row[accession_index])
-                length = len(accessions)
+def read_csv():
+    global upload_folder
+    upload_file = upload_folder + csv_file # Change the storage folder path
```

Comment:
Here too, from the commandline execution option,
"csv_file" is an existing complete value when executed directly from the cold-extraction module.

The csv_file takes input from the config.json as a full path.
csv_file = args['CsvFile']
(When we run from the command-line, it is folder-name/file-name already. Please check the sample in config.json)
So, 
upload_file = upload_folder + csv_file will actually fail, by introducing the folder name two times when run directly.

So, my suggestion is:
rather than passing a file name and folder name separately, perhaps, compose the full csv_file path in the front-end module (to end up in the cold-extraction/csv folder), and pass it here.

That way, you don't need custom code for cold-extraction run directly from commandline vs from the front-end.